test/integration/tests/hierarchycontrol.sh

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+source helpers.sh
+
+cleanup() {
+
+  if [ "$1" != "no-shut-down" ]; then
+      shut_down
+  fi
+}
+trap cleanup EXIT
+
+start_up
+
+cleanup "no-shut-down"
+
+tpm2_clear
+
+# Clear the handler
+trap - ERR
```

Comment:
reverse the order of the tests so you just do normal testing up front, then clear the trap handler, then do negative testing.

Reply:
Thank you for this comment, I will change the testing flow.